File: prototype/Player.hs
module Player (Player (Blue, Orange), other) where

------------------------------------------------------------------------------

data Player = Blue | Orange deriving (Eq, Ord)

other :: Player -> Player
other Blue = Orange
other Orange = Blue

instance Show Player where
  showsPrec _ Blue = (++) "B"
  showsPrec _ Orange = (++) "O"

File: player.cpp
// mirko.rahn@web.de

#include <player.hpp>

namespace
{
  namespace player
  {
    player other (player p)
    {
      return (p == Blue) ? Orange : Blue;
    }

    show::show (player const& player)
      : _player (player)
    {}
    std::ostream& show::operator() (std::ostream& os) const
    {
      return os << ( (_player == Blue) ? 'B'
                   : (_player == Orange) ? 'O'
                   : '.'
                   );
    }
  }
}
